### PR TITLE
[OpenCL][CI] Enable OpenCL cpp tests in CI

### PR DESCRIPTION
**Pull request by echuraev**
Enable OpenCL cpp tests in CI

### CODE REVIEW

**Comment by echuraev** [comment]
@driazati, @masahi please review it.

File: tests/scripts/task_config_build_gpu.sh
Comment (echuraev) [inline tests/scripts/task_config_build_gpu.sh:39]: It is necessary to have google tests source code to build them for the device. In the CI we could use system GTest, but I didn't want to change the build mechanism in the CMake files, because I think it is better not to use system gtest but build it for the target device. 
This is why we need to download gtest source code in the build directory and build `tvm_runtime` with it.

Comment (driazati) [reply tests/scripts/task_config_build_gpu.sh:39]: The downloading is fine but this should be moved into the GPU docker image (https://github.com/apache/tvm/blob/main/docker/Dockerfile.ci_gpu) so we don't have to download it fresh each time. You'll have to send a separate PR for the docker changes, then we can update the images to use the new one, then once that's done we can merge this PR

**Review by driazati — APPROVED** [review]

**Comment by echuraev** [comment]
@tvm-bot rerun

File: ci/jenkins/templates/gpu_jenkinsfile.groovy.j2
Comment (driazati) [inline ci/jenkins/templates/gpu_jenkinsfile.groovy.j2:84]: Can you move this next to the `cpp_unittests` above? Right now this job is run 3 times afaik and pytest knows that each job should only run 1/3 of the tests, but this runs all the OpenCL tests 3 times

Comment (echuraev) [reply ci/jenkins/templates/gpu_jenkinsfile.groovy.j2:84]: Done

**Review by driazati — APPROVED** [review]

**Comment by driazati** [comment]
lgtm assuming CI passes on the /branch jobs

**Pull request merged**
